### PR TITLE
Adds a rate-limited progress when exporting data from process logs

### PR DESCRIPTION
**Pull request by idlira**

### CODE REVIEW

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java [hunk 1]
```diff
@@ -225,4 +241,10 @@ private void writeTableRow(List<String> columns, LineBasedExport export, Process
                             .handle();
         }
     }
+
+    private void setExportState(ProcessContext processContext, int currentRow, boolean lastCall) {
+        if (exportLimiter.check() || lastCall) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java:246]:
```suggestion
        if (TaskContext.get().shouldUpdateState() || lastCall) {
```

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java [hunk 2]
```diff
@@ -79,6 +83,8 @@
     @Part
     private TableProcessOutputType tableProcessOutputType;
 
+    private final RateLimit exportLimiter = RateLimit.timeInterval(10, TimeUnit.SECONDS);
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java:86]:
```suggestion
```

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java [hunk 3]
```diff
@@ -38,6 +40,8 @@
 import java.io.OutputStream;
 import java.io.OutputStreamWriter;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java:43]:
```suggestion
```

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java [hunk 4]
```diff
@@ -225,4 +241,10 @@ private void writeTableRow(List<String> columns, LineBasedExport export, Process
                             .handle();
         }
     }
+
+    private void setExportState(ProcessContext processContext, int currentRow, boolean lastCall) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java:245]:
```suggestion
    private void updateExportState(ProcessContext processContext, int currentRow, boolean lastCall) {
```

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java [hunk 5]
```diff
@@ -207,7 +218,12 @@ private void exportTable(ExportFileType exportFileType, ProcessOutput out, Proce
             List<String> columns = tableProcessOutputType.determineColumns(out);
             List<String> labels = tableProcessOutputType.determineLabels(out, columns);
             export.addRow(labels);
-            createLogsQuery(out, processContext).iterateAll(logEntry -> writeTableRow(columns, export, logEntry));
+            AtomicInteger rowCount = new AtomicInteger(0);
+            createLogsQuery(out, processContext).iterateAll(logEntry -> {
+                writeTableRow(columns, export, logEntry);
+                setExportState(processContext, rowCount.incrementAndGet(), false);
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java:224]: or simply processContext.incCounter("$Process.rowsExported");

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java [hunk 6]
```diff
@@ -146,7 +152,12 @@ private void exportLogs(ExportFileType exportFileType, @Nullable ProcessOutput o
                                NLS.get("ProcessLog.message"),
                                NLS.get("ProcessLog.messageType"),
                                NLS.get("ProcessLog.node"));
-            createLogsQuery(out, processContext).iterateAll(logEntry -> writeLog(export, logEntry));
+            AtomicInteger rowCount = new AtomicInteger(0);
+            createLogsQuery(out, processContext).iterateAll(logEntry -> {
+                writeLog(export, logEntry);
+                setExportState(processContext, rowCount.incrementAndGet(), false);
```
Comment (andyHa) [inline src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java:158]: see below

File: src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java (hunk 5 above)
Comment (idlira) [reply src/main/java/sirius/biz/process/ExportLogsAsFileTaskExecutor.java]: I found the state "nicer" then just a count. The state can also be seen from the job list view.

**Review by sabieber — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Review by ymo-sci — APPROVED** [review]

**Pull request merged**